Issue with Nuka data and standards

Pretty normal data set covering multiple days with standard records. For some reason they're not being found for STD2:

```
uk.ac.exeter.QuinCe.jobs.JobFailedException: Job ID 717 failed
	at uk.ac.exeter.QuinCe.jobs.files.DataReductionJob.execute(DataReductionJob.java:229)
	at uk.ac.exeter.QuinCe.jobs.JobThread.run(JobThread.java:79)
Caused by: uk.ac.exeter.QuinCe.data.Dataset.DataReduction.DataReductionException: Error while applying internal calibration
	at uk.ac.exeter.QuinCe.data.Dataset.DataReduction.DataReducer.applyValueCalibration(DataReducer.java:397)
	at uk.ac.exeter.QuinCe.data.Dataset.DataReduction.UnderwayMarinePco2Reducer.doCalculation(UnderwayMarinePco2Reducer.java:78)
	at uk.ac.exeter.QuinCe.data.Dataset.DataReduction.DataReducer.performDataReduction(DataReducer.java:113)
	at uk.ac.exeter.QuinCe.jobs.files.DataReductionJob.execute(DataReductionJob.java:169)
	... 1 more
Caused by: uk.ac.exeter.QuinCe.utils.RecordNotFoundException: No calibrations found for external standard 'std2'
	at uk.ac.exeter.QuinCe.data.Dataset.DataReduction.DataReducer.calculateStandardValueAtDate(DataReducer.java:425)
	at uk.ac.exeter.QuinCe.data.Dataset.DataReduction.DataReducer.applyValueCalibration(DataReducer.java:381)
	... 4 more
```

--- a/WebApp/src/uk/ac/exeter/QuinCe/jobs/files/ExtractDataSetJob.java
+++ b/WebApp/src/uk/ac/exeter/QuinCe/jobs/files/ExtractDataSetJob.java
@@ -285,9 +285,13 @@ public class ExtractDataSetJob extends Job {
 
     boolean result = false;
 
-    if (DateTimeUtils.secondsBetween(runTypePeriod.start, time) <= instrument.getPreFlushingTime()) {
+    if (instrument.getPreFlushingTime() > 0 &&
+      DateTimeUtils.secondsBetween(runTypePeriod.start, time) <= instrument.getPreFlushingTime()) {
+
       result = true;
-    } else if (DateTimeUtils.secondsBetween(time, runTypePeriod.end) <= instrument.getPostFlushingTime()) {
+    } else if (instrument.getPostFlushingTime() > 0 &&
+      DateTimeUtils.secondsBetween(time, runTypePeriod.end) <= instrument.getPostFlushingTime()) {
+
       result = true;
     }
 

